### ISSUE TITLE
Question about ESPHome OpenTherm

### ISSUE THREAD

**Issue by foxalex85** [bluekiller/homemate_bridge#1]
Hi, sorry if I write here, but I didn’t found another way.

I’ve tried your ESPHome OpenTherm, that was modified with async calls, but it seems not work.
With the original repository by risciano I’ve successfully connected with the boiler, while with this version there are somethings problem.

Did you have these problems or in your test all worked?

Thanks

**Comment by bluekiller** [comment]
I can connect to my boiler(Viessmann), but my boiler seems not supporting opentherm well. Did you change the **_mock_** variable @line 15 in opentherm_component.h to false?

`bool mock = true;
`

my log
[17:49:59][D][opentherm_component:163]: Domestic hot water temperature is 29.000000 °C
[17:49:59][D][climate:385]: 'Domestic Hot Water' - Sending state:
[17:49:59][D][climate:388]:   Mode: HEAT
[17:49:59][D][climate:390]:   Action: OFF
[17:49:59][D][climate:408]:   Current Temperature: 29.00°C
[17:49:59][D][climate:414]:   Target Temperature: 42.00°C
[17:49:59][D][opentherm_component:267]: Opentherm API is not ready
[17:50:00][D][opentherm_component:197]: Request timeout
[17:50:00][D][opentherm_component:388]: Getting domestic hot water flow rate...
[17:50:01][D][opentherm_component:172]: Domestic hot water flow rate is 52.000000 L/min
[17:50:01][D][sensor:124]: 'Hot Water Flow Rate': Sending state 52.00000 L/min with 1 decimals of accuracy
[17:50:01][D][opentherm_component:267]: Opentherm API is not ready
[17:50:02][D][opentherm_component:197]: Request timeout
[17:50:02][D][opentherm_component:285]: Setting central heating status: OFF
[17:50:02][D][opentherm_component:286]: Setting domestic hot water status: ON
[17:50:03][D][opentherm_component:115]: Central Heating: OFF
[17:50:03][D][opentherm_component:116]: Hot Water: OFF
[17:50:03][D][opentherm_component:117]: Flame: ON
[17:50:03][D][binary_sensor:036]: 'Boiler Flame': Sending state ON
[17:50:03][D][climate:385]: 'Domestic Hot Water' - Sending state:
[17:50:03][D][climate:388]:   Mode: HEAT
[17:50:03][D][climate:390]:   Action: OFF
[17:50:03][D][climate:408]:   Current Temperature: 29.00°C
[17:50:03][D][climate:414]:   Target Temperature: 42.00°C
[17:50:03][D][climate:385]: 'Central Heating' - Sending state:
[17:50:03][D][climate:388]:   Mode: OFF
[17:50:03][D][climate:390]:   Action: OFF
[17:50:03][D][climate:408]:   Current Temperature: 29.00°C
[17:50:03][D][climate:411]:   Target Temperature: Low: 20.00°C High: 65.00°C
[17:50:03][D][opentherm_component:267]: Opentherm API is not ready
[17:50:04][D][opentherm_component:197]: Request timeout
[17:50:04][D][opentherm_component:333]: Setting domestic hot water temperature at 42.000000 °C...
[17:50:05][D][opentherm_component:199]: Response invalid: 0xF0382A00
[17:50:05][D][opentherm_component:267]: Opentherm API is not ready
[17:50:06][D][opentherm_component:197]: Request timeout
[17:50:06][D][opentherm_component:333]: Setting domestic hot water temperature at 42.000000 °C...
[17:50:07][D][opentherm_component:199]: Response invalid: 0xF0382A00
[17:50:07][D][opentherm_component:267]: Opentherm API is not ready
[17:50:08][D][opentherm_component:197]: Request timeout
[17:50:08][D][opentherm_component:333]: Setting domestic hot water temperature at 42.000000 °C...
[17:50:09][D][opentherm_component:199]: Response invalid: 0xF0382A00
[17:50:09][D][opentherm_component:267]: Opentherm API is not ready
[17:50:10][D][opentherm_component:197]: Request timeout
[17:50:10][D][opentherm_component:378]: Getting domestic hot water temperature...
[17:50:11][D][opentherm_component:163]: Domestic hot water temperature is 42.000000 °C
[17:50:11][D][climate:385]: 'Domestic Hot Water' - Sending state:
[17:50:11][D][climate:388]:   Mode: HEAT
[17:50:11][D][climate:390]:   Action: OFF
[17:50:11][D][climate:408]:   Current Temperature: 42.00°C
[17:50:11][D][climate:414]:   Target Temperature: 42.00°C
[17:50:11][D][opentherm_component:267]: Opentherm API is not ready
[17:50:12][D][opentherm_component:197]: Request timeout
[17:50:12][D][opentherm_component:388]: Getting domestic hot water flow rate...
[17:50:13][D][opentherm_component:172]: Domestic hot water flow rate is 51.000000 L/min
[17:50:13][D][sensor:124]: 'Hot Water Flow Rate': Sending state 51.00000 L/min with 1 decimals of accuracy
[17:50:13][D][opentherm_component:267]: Opentherm API is not ready
[17:50:14][D][opentherm_component:197]: Request timeout
[17:50:14][D][opentherm_component:285]: Setting central heating status: OFF
[17:50:14][D][opentherm_component:286]: Setting domestic hot water status: ON
[17:50:15][D][opentherm_component:115]: Central Heating: OFF
[17:50:15][D][opentherm_component:116]: Hot Water: OFF
[17:50:15][D][opentherm_component:117]: Flame: ON

**Comment by foxalex85** [comment]
Yes, finally I was able to connect to the boiler. For my boiler not all commands are supported. For example: I cannot set the domestic hot water and I cannot retrieve the flow rate. The "Opentherm API is not ready" in the log I thing is normal, because the OpenTherm protocol seems to be a little bit slow. If you want to check the boiler compatibility matrix you can check this page http://otgw.tclcode.com/matrix.cgi#boilers
